Add Nix Support

404Wolf opened the pull request:
TLDR: If you have nix, now you'll be able to just do `nix run github:cloudsftp/reSnap` to run the script with all dependencies automatically fetched and "baked" into it.

I added a basic nix flake and package that includes `feh`, `ffmpeg`, `lz4`, and `imagemagick`. It's pretty simple but makes it easier to use for anyone who has nix installed.

cloudsftp commented:
Sorry, this would be better in its own Repo, I don't really want to maintain it.

A nix package, that is installable with one line in the config on the other hand would actually be nice.
If you build something like that, we can put it in the README

404Wolf commented:
> Sorry, this would be better in its own Repo, I don't really want to maintain it.
> 
> A nix package, that is installable with one line in the config on the other hand would actually be nice. If you build something like that, we can put it in the README

Not quite sure I understand what you mean. You mean PR this to nixpkgs? I'm happy to do that -- that's a good idea. The way I packaged it isn't  nixos specific/doesn't need nixos.

The way that I've added the nix support here is really just listing the dependencies by name (to create a shell script with access to them), at the least I feel like it would be helpful for developers to have a nix shell with the dependencies, instead of making everyone install all of them globally. Would you be open to me watering down the two files I added to just be a flake with a devshell?

Thanks for reviewing!

cloudsftp commented:
yes, exactly. and consider hosting a seperate repo for the nix stuff importing this via git submodules or smth.

I don't know nix very well and am trying to keep this repo slim.

Also i think you revert #17 with this pr

sorry for not merging
